scripts/esbuild/testing.ts

```diff
@@ -0,0 +1,141 @@
+import type { BuildOptions as ESBuildOptions, Plugin } from 'esbuild';
+import fs from 'fs-extra';
+import { join } from 'path';
+
+import { copyTestingInternalDts } from '../bundles/testing';
+import { getBanner } from '../utils/banner';
+import type { BuildOptions } from '../utils/options';
+import { writePkgJson } from '../utils/write-pkg-json';
+import { getBaseEsbuildOptions, getEsbuildAliases, getEsbuildExternalModules, runBuilds } from './util';
+
+const EXTERNAL_TESTING_MODULES = [
+  'assert',
+  'buffer',
+  'child_process',
+  'console',
+  'constants',
+  'crypto',
+  'fs',
+  '@jest/core',
+  'jest-cli',
+  'jest',
+  'expect',
+  '@jest/reporters',
+  'jest-environment-node',
+  'jest-message-id',
+  'jest-runner',
+  'net',
+  'os',
+  'path',
+  'process',
+  'puppeteer',
+  'puppeteer-core',
+  'readline',
+  'rollup',
+  '@rollup/plugin-commonjs',
+  '@rollup/plugin-node-resolve',
+  'stream',
+  'tty',
+  'url',
+  'util',
+  'vm',
+  'yargs',
+  'zlib',
+];
+
+export async function buildTesting(opts: BuildOptions) {
+  const inputDir = join(opts.buildDir, 'testing');
+  const sourceDir = join(opts.srcDir, 'testing');
+
+  await Promise.all([
+    // copy jest testing entry files
+    fs.copy(join(opts.scriptsBundlesDir, 'helpers', 'jest'), opts.output.testingDir),
+    copyTestingInternalDts(opts, inputDir),
+  ]);
+
+  // write package.json
+  writePkgJson(opts, opts.output.testingDir, {
+    name: '@stencil/core/testing',
+    description: 'Stencil testing suite.',
+    main: 'index.js',
+    types: 'index.d.ts',
+  });
+
+  const external = [
+    ...EXTERNAL_TESTING_MODULES,
+    ...getEsbuildExternalModules(opts, opts.output.testingDir),
+    '@rollup/plugin-commonjs',
+    '@rollup/plugin-node-resolve',
+    'rollup',
+    '../compiler/stencil.js',
+  ];
+
+  const testingEsbuildOptions: ESBuildOptions = {
+    ...getBaseEsbuildOptions(),
+    entryPoints: [join(sourceDir, 'index.ts')],
+    bundle: true,
+    format: 'cjs',
+    outfile: join(opts.output.testingDir, 'index.js'),
+    platform: 'node',
+    logLevel: 'info',
+    external,
+    write: false,
+    alias: getEsbuildAliases(),
+    banner: { js: getBanner(opts, `Stencil Testing`, true) },
+    plugins: [
+      externalAliases('@app-data', '@stencil/core/internal/app-data'),
+      externalAliases('@platform', '@stencil/core/internal/testing'),
+      externalAliases('../internal/testing/index.js', '@stencil/core/internal/testing'),
+      externalAliases('@stencil/core/dev-server', '../dev-server/index.js'),
+      lazyRequirePlugin(opts, [
+        '@stencil/core/internal/app-data',
+        '@stencil/core/internal/testing',
+        '../dev-server/index.js',
+        '../internal/testing/index.js',
+        '../mock-doc/index.cjs',
+      ]),
+    ],
```

Comment:
I'm not sure I fully understand the differences between this section and what we have in `bundles/testing.ts` today https://github.com/ionic-team/stencil/blob/865ae1d36e4d436d9011ddfb09690d41a7709684/scripts/bundles/testing.ts#L80-L83

Specifically: Why do we not externalize `@stencil/core/mock-doc` to `../mock-doc/index.cjs` (like we do for the ESBuild scripts)? Do we no longer need this?

Reply:
Update: I was assuming that `mock-doc` was externalised but further investigation has shown that it is part of the bundle. I debugged through the build to find why that is and hope to find a resolution today.